src/test/groovy/tests/DatadogTest.groovy

```diff
@@ -386,8 +360,8 @@ class DatadogTest extends DataAnalyticsSteps {
         def from_timestamp = (now.getTime()-1800000).toString().substring(0,10)
         def to_timestamp = (now.getTime()).toString().substring(0,10)
         def query = "sum:active_uploading_ips{*} by {remote_address}.as_count()"
-        Response response = datadog.datadogQueryTimeSeriesPoints(datadog_url,
-                datadog_api_key, datadog_application_key, from_timestamp, to_timestamp, query)
+        Response response = datadog.datadogQueryTimeSeriesPoints(datadogBaseURL,
+                datadogApiKey, datadogApplicationKey, from_timestamp, to_timestamp, query)
         def IPv4andIPv6Regex = "(^remote_address:)(([a-za-z:])([0-9A-Fa-f]{1,4}:){7}[0-9A-Fa-f]{1,4}|(\\d{1,3}\\.){3}\\d{1,3})"
```

Comment:
OH! Much easier way to do this and be more reliable!

```
groovy:000> import java.net.InetAddress
===> java.net.InetAddress
groovy:000> InetAddress.getByName('127.0.0.1')
===> /127.0.0.1
groovy:000> InetAddress.getByName('192.168.1.1')
===> /192.168.1.1
groovy:000> InetAddress.getByName('blah')
ERROR java.net.UnknownHostException:
blah: nodename nor servname provided, or not known
        at java_net_InetAddress$getByName.call (Unknown Source)
groovy:000> InetAddress.getByName('www.google.com')
===> www.google.com/216.58.217.36
groovy:000> InetAddress.getByName('294.197.1.1')
ERROR java.net.UnknownHostException:
294.197.1.1: nodename nor servname provided, or not known
        at java_net_InetAddress$getByName.call (Unknown Source)
groovy:000> 

```

in the event the address/host isn't valid, you get an exception

Reply:
Regex matches the text starting with "remote_address: " and followed by IP address. So I feel that regex suits better.

Reply:
Ah, I didn't notice it had something in front of it. I think you're still better off using regex groups to remove the prefix and then using _very_ robust java libraries to validate a host name.